nextroute/schema/fleet.go

```diff
@@ -0,0 +1,234 @@
+// Package schema provides the input and output schema for nextroute.
+package schema
+
+import (
+	"fmt"
+	"time"
+)
+
+// FleetInput schema.
+type FleetInput struct {
+	Defaults       *FleetDefaults  `json:"defaults,omitempty"`
+	Vehicles       []FleetVehicle  `json:"vehicles,omitempty"`
+	Stops          []FleetStop     `json:"stops,omitempty"`
+	StopGroups     [][]string      `json:"stop_groups,omitempty"`
+	AlternateStops []FleetStop     `json:"alternate_stops,omitempty"`
+	DurationGroups []DurationGroup `json:"duration_groups,omitempty"`
+}
+
+// FleetDefaults holds the fleet input default data.
+type FleetDefaults struct {
+	Vehicles *FleetVehicleDefaults `json:"vehicles,omitempty"`
+	Stops    *FleetStopDefaults    `json:"stops,omitempty"`
+}
+
+// FleetVehicleDefaults holds the fleet input vehicle default data.
+type FleetVehicleDefaults struct {
```

Comment:
So on so forth about adding the deprecation notice. You just need to change the correct struct on each comment to point to `[VehicleDefaults]`, `[StopDefaults]`, etc...

Reply:
followed it except for the solve options because there are so many nested structs that do not have a direct match. so its only "solve.Options" for those cases.